codegen/smithy-go-codegen/src/main/java/software/amazon/smithy/go/codegen/ShapeValueGenerator.java

```diff
@@ -0,0 +1,367 @@
+        public Void arrayNode(ArrayNode node) {
+            Shape memberShape = model.expectShape(((CollectionShape) this.currentShape).getMember().getTarget());
+
+            node.getElements().forEach(element -> {
+                valueGen.writeShapeValueInline(writer, memberShape, element);
+                writer.write(",");
+            });
+            return null;
+        }
+
+        @Override
+        public Void objectNode(ObjectNode node) {
+            node.getMembers().forEach((keyNode, valueNode) -> {
+                Shape memberShape;
+                switch (currentShape.getType()) {
+                    case STRUCTURE:
+                        MemberShape member;
+                        if (currentShape.asStructureShape().get().getMember(keyNode.getValue()).isPresent()) {
+                            member = currentShape.asStructureShape().get().getMember(keyNode.getValue()).get();
+                        } else {
+                            throw new CodegenException(
+                                    "unknown member " + currentShape.getId() + "." + keyNode.getValue());
+                        }
+
+                        memberShape = model.expectShape(member.getTarget());
+                        String memberName = symbolProvider.toMemberName(member);
+
+                        writer.write("$L: ", memberName);
+                        valueGen.writeShapeValueInline(writer, memberShape, valueNode);
+                        writer.write(",");
+                        break;
+
+                    case MAP:
+                        memberShape = model.expectShape(this.currentShape.asMapShape().get().getValue().getTarget());
+
+                        writer.write("$S: ", keyNode.getValue());
+                        valueGen.writeShapeValueInline(writer, memberShape, valueNode);
+                        writer.write(",");
+                        break;
+
+                    default:
+                        throw new CodegenException("unexpected shape type " + currentShape.getType());
+                }
+            });
+
+            return null;
+        }
+
+        @Override
+        public Void booleanNode(BooleanNode node) {
+            if (!currentShape.getType().equals(ShapeType.BOOLEAN)) {
+                throw new CodegenException("unexpected shape type " + currentShape + " for boolean value");
+            }
+
+            writer.writeInline("$L", node.getValue() ? "true" : "false");
+            return null;
+        }
+
+        @Override
+        public Void nullNode(NullNode node) {
+            if (currentShape.getType() == ShapeType.STRING && currentShape.hasTrait(EnumTrait.class)) {
+                writer.writeInline("\"\"");
+            } else {
+                writer.writeInline("nil");
+            }
+
+            return null;
+        }
+
+        @Override
+        public Void numberNode(NumberNode node) {
+            switch (currentShape.getType()) {
+                case BOOLEAN:
```

Comment:
The non-[number | timestamp] shape types won't have number nodes representing them. Smithy validates this for the params property of each of the test traits.